server/src/inmantals/server.py

```diff
@@ -399,27 +405,44 @@ def get_range_from_position(self, textDocument, position)-> Optional[Interval]:
             return None
         return range
 
+    def get_range_from_position_reverse(self, textDocument, position) -> Optional[Interval]:
```

Comment:
There's a lot of duplication between this method and the one above. I think it could be simplified a lot by extracting this into a separate method that just accepts the dict to do the lookup in.